modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/OptionCategoryResourceImpl.java

```diff
@@ -98,36 +123,37 @@ public Response patchOptionCategory(Long id, OptionCategory optionCategory)
 	}
 
 	@Override
-	public OptionCategory postOptionCategory(OptionCategory optionCategory)
+	public Response patchOptionCategoryByExternalReferenceCode(
+			String externalReferenceCode, OptionCategory optionCategory)
 		throws Exception {
 
-		CPOptionCategory cpOptionCategory = null;
+		CPOptionCategory cpOptionCategory =
+			_cpOptionCategoryService.getCPOptionCategoryByExternalReferenceCode(
+				externalReferenceCode, contextCompany.getCompanyId());
 
-		if (optionCategory.getId() != null) {
-			cpOptionCategory = _cpOptionCategoryService.fetchCPOptionCategory(
-				optionCategory.getId());
-		}
+		return patchOptionCategory(
+			cpOptionCategory.getCPOptionCategoryId(), optionCategory);
+	}
 
-		if (cpOptionCategory == null) {
-			cpOptionCategory = _addOptionCategory(optionCategory);
-		}
-		else {
-			cpOptionCategory = _updateOptionCategory(
-				optionCategory.getId(), optionCategory);
-		}
+	@Override
+	public OptionCategory postOptionCategory(OptionCategory optionCategory)
+		throws Exception {
 
-		return _toOptionCategory(cpOptionCategory.getCPOptionCategoryId());
+		return _toOptionCategory(
+			_cpOptionCategoryService.addOrUpdateCPOptionCategory(
+				GetterUtil.getString(optionCategory.getExternalReferenceCode()),
+				GetterUtil.getLong(optionCategory.getId()),
+				LanguageUtils.getLocalizedMap(optionCategory.getTitle()),
+				LanguageUtils.getLocalizedMap(optionCategory.getDescription()),
+				GetterUtil.getLong(optionCategory.getPriority()),
+				optionCategory.getKey(),
+				_serviceContextHelper.getServiceContext(contextUser)));
 	}
 
-	private CPOptionCategory _addOptionCategory(OptionCategory optionCategory)
+	private OptionCategory _toOptionCategory(CPOptionCategory cpOptionCategory)
```

Comment:
is really needed? Can we pass directly the id the `_toOptionCategory(Long cpOptionCategoryId)`?